File: archaic/src/main/scala/resonantinduction/archaic/firebox/RenderHotPlate.java
package resonantinduction.archaic.firebox;

import net.minecraft.client.renderer.RenderBlocks;
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
import net.minecraft.tileentity.TileEntity;
import net.minecraftforge.common.ForgeDirection;
import resonant.lib.render.RenderItemOverlayUtility;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

@SideOnly(Side.CLIENT)
public class RenderHotPlate extends TileEntitySpecialRenderer
{
	private final RenderBlocks renderBlocks = new RenderBlocks();

	@Override
	public void renderTileEntityAt(TileEntity tileEntity, double x, double y, double z, float var8)
	{
		if (tileEntity instanceof TileHotPlate)
		{
			TileHotPlate tile = (TileHotPlate) tileEntity;
			RenderItemOverlayUtility.renderTopOverlay(tileEntity, tile.getInventory().getContainedItems(), ForgeDirection.EAST, 2, 2, x, y - 0.8, z, 1f);
		}
	}
}

File: basic/main/scala/com/calclavia/edx/basic/process/smelting/firebox/RenderHotPlate.scala
package com.calclavia.edx.basic.process.smelting.firebox

import cpw.mods.fml.relauncher.{Side, SideOnly}
import net.minecraft.client.renderer.RenderBlocks
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer
import net.minecraft.tileentity.TileEntity
import net.minecraftforge.common.util.ForgeDirection
import resonantengine.lib.render.RenderItemOverlayUtility

@SideOnly(Side.CLIENT) class RenderHotPlate extends TileEntitySpecialRenderer
{
  private final val renderBlocks: RenderBlocks = new RenderBlocks

  def renderTileEntityAt(tileEntity: TileEntity, x: Double, y: Double, z: Double, var8: Float)
  {
    if (tileEntity.isInstanceOf[TileHotPlate])
    {
      val tile: TileHotPlate = tileEntity.asInstanceOf[TileHotPlate]
      RenderItemOverlayUtility.renderTopOverlay(tileEntity, tile.getInventory.getContainedItems, ForgeDirection.EAST, 2, 2, x, y - 0.8, z, 1f)
    }
  }

}